File: Lazy/Lazy/ILazy.cs
namespace Lazy
{
    /// <summary>
    /// Интерфейс ILazy представляет ленивые вычисления.
    /// </summary>
    /// <typeparam name="T">Тип вычисления, возвращаемый ленивой функцией.</typeparam>
    public interface ILazy<T>
    {
        T Get { get; }  
    }
}

File: sem 4/Exercise 8/Exercise 8/Lazy/ILazy.fs
namespace Lazy

module ILazy =

    // Интерфейс леинвых вычислений. 
    // 'a -- тип вычисления, возвращаемого ленивой функцией.
    type ILazy<'a> =

        // Представляет результат ленивого вычисления.
        abstract member Get: unit -> 'a


